File: src/main/org/nlogo/prim/_not.java
package org.nlogo.prim;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _not
    extends Reporter
    implements org.nlogo.nvm.Pure {
  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax
        (new int[]{Syntax.TYPE_BOOLEAN},
            Syntax.TYPE_BOOLEAN);
  }

  @Override
  public Object report(Context context)
      throws LogoException {
    return report_1(context, argEvalBooleanValue(context, 0));
  }

  public boolean report_1(Context context, boolean arg0) {
    return !arg0;
  }
}

File: src/main/org/nlogo/prim/_not.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Reporter, Pure }

class _not extends Reporter with Pure {

  override def syntax =
    Syntax.reporterSyntax(
      Array(Syntax.BooleanType),
      Syntax.BooleanType)

  override def report(context: Context): java.lang.Boolean =
    Boolean.box(report_1(context, argEvalBooleanValue(context, 0)))

  def report_1(context: Context, arg0: Boolean): Boolean =
    !arg0

}
